Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h

```diff
@@ -214,6 +217,7 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
 
     WebCore::IntPoint convertFromPluginToDocument(const WebCore::IntPoint&) const;
     std::optional<PDFDocumentLayout::PageIndex> nearestPageIndexForDocumentPoint(const WebCore::IntPoint&) const;
+    std::optional<RetainPtr<PDFAnnotation>> annotationForRootViewPoint(const WebCore::IntPoint&) const;
```

Comment:
A pointer type has nullness built in, do we need to wrap it in `std::optional`?

Reply:
or do you intend for there to be a semantic difference between the `std::nullopt` and `nullptr` return values here?

Reply:
Yeah I was probably overcomplicating it. I wanted to really make sure the caller checks that the `RetainPtr` could very easily be null in this case, but the function definition makes it clear enough